snmp/datadog_checks/snmp/commands.py

```diff
@@ -110,6 +116,9 @@ def snmp_bulk(config, oid, non_repeaters, max_repetitions, lookup_mib, ignore_no
     # type: (InstanceConfig, hlapi.ObjectType, int, int, bool, bool) -> Generator
     """Call SNMP GETBULK on an oid."""
 
+    if config.device is None:
```

Comment:
Can that happen ? Not sure if it's needed.

Reply:
This is mostly a mypy hint, and _technically_ it could yes (depending on how/where we're calling these functions), so I think it's better to make that assumption clear. My hope is that further refactoring will make these checks irrelevant (eg by passing `target` directly).